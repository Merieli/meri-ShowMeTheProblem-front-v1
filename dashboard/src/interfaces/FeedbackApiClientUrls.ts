export interface FeedbackApiClientUrls {
    create: string
    show: string
    getFilters: string
}
