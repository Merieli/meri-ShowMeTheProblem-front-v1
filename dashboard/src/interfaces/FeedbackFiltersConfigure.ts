import { IFeedbackFilters, TFeedback } from './'

export interface FeedbackFiltersConfigure {
    data: IFeedbackFilters
    typeActive: TFeedback
}
