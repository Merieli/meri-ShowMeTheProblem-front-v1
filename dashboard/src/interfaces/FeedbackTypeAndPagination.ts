import { TFeedback } from './'

export interface FeedbackTypeAndPagination {
    type: TFeedback
    limit: number
    offset: number
}
