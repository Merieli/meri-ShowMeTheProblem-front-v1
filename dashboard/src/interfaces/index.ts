import { IAccount } from './IAccount'

export * from './'
