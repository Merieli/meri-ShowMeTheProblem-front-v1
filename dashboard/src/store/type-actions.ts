export enum Actions {
    LOGIN_USER = 'LOGIN_USER',
    REGISTER_USER = 'REGISTER_USER',
    GET_USER = 'GET_USER',

    NEW_FEEDBACK = 'NEW_FEEDBACK',
    GET_ALL_FEEDBACKS = 'GET_ALL_FEEDBACKS',
    GET_FEEDBACKS = 'GET_FEEDBACKS',
    GET_INDEX_FEEDBACK = 'GET_INDEX_FEEDBACK',
    CHANGE_ACTIVE_FEEDBACK = 'CHANGE_ACTIVE_FEEDBACK',
}
