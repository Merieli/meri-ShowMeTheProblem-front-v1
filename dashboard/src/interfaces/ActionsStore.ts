import { FeedbackTypeAndPagination, IUser, StateStoreShape, TFeedback } from '.';
import { Actions } from '@/store/type-actions';
import { ActionContext, ActionTree } from 'vuex';

type Context = ActionContext<StateStoreShape, StateStoreShape>;

export interface ActionsStore extends ActionTree<StateStoreShape, StateStoreShape> {
    /**
     * @name REGISTER_USER
     * @descripton efetua o registro do usuário e o adiciona a store
     * @param {IUser} user usuário com name, email e password
     */
    [Actions.REGISTER_USER]: ({ commit }: Context, user: IUser) => Promise<void>;

    /**
     * @name LOGIN_USER
     * @description Efetua login do usuário, salva o token na store,
     * redireciona para a página de 'feedbacks' e aciona a action
     * para pegar os dados do usuário logado.
     * @param {IUser} user usuário com name, email e password
     */
    [Actions.LOGIN_USER]: (context: Context, user: IUser) => Promise<void>;

    /**
     * @name GET_USER
     * @descripton Captura os dados do usuário e efetua o login se já tiver um
     * token de usuário salvo, armazenando os dados na store
     */
    [Actions.GET_USER]: (context: Context) => Promise<void>;

    /**
     * @name LOGGOUT_USER
     * @descripton faz o usuário ser deslogado
     */
    [Actions.LOGGOUT_USER]: (context: Context) => Promise<void>;

    /**
     * @name GENERATE_NEW_KEY
     * @descripton gera uma nova chave de api para o usuário e a salva na store
     */
    [Actions.GENERATE_NEW_KEY]: (context: Context) => Promise<void>;

    [Actions.HANDLE_COPY_TEXT]: (context: Context, text: string) => Promise<void>;

    /**
     * @name GET_SUMMARY_FEEDBACK
     * @descripton pega a quantidade de cada tipo de feedback e salva na store
     * configurando para exibir os filtros da listagem
     */
    [Actions.GET_SUMMARY_FEEDBACK]: (context: Context) => Promise<void>;

    /**
     * @name CHANGE_ACTIVE_FEEDBACK
     * @descripton altera o tipo de feedback ativo
     * @param {TFeedback} type tipo do feedback retornado
     */
    [Actions.CHANGE_ACTIVE_FEEDBACK]: (context: Context, type: TFeedback) => Promise<void>;

    /**
     * @name GET_ALL_FEEDBACKS
     * @descripton
     * @param {Object} paramsRequest
     */
    [Actions.GET_ALL_FEEDBACKS]: (
        context: Context,
        feedbackTypeAndPagination: FeedbackTypeAndPagination
    ) => Promise<void>;
}
