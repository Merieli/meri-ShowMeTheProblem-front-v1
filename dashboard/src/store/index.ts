import useNotifier from '../hooks/notifier'
import router from '../router/index'
import { Actions } from './type-actions'
import { Mutations } from './type-mutations'
import callApiClient from '@/http-client/index'
import { TypeOfNotification } from '@/interfaces/INotification'
import { INotification } from '@/interfaces/INotification'
import IUser from '@/interfaces/IUser'
import IUserLogged from '@/interfaces/IUserLogged'
import { createStore } from 'vuex'

export interface EstadoStore {
    users: IUser[]
    notifications: INotification[]
    isLogged: boolean
    userLogged: IUserLogged
}

export const store = createStore<EstadoStore>({
    state: {
        users: [],
        notifications: [],
        isLogged: false,
        userLogged: {
            name: '',
            token: '',
            apiKey: '',
        },
    },
    getters: {},
    mutations: {
        [Mutations.NOTIFY](state, newNotification: INotification) {
            newNotification.id = new Date().getTime()
            state.notifications.push(newNotification)
            setTimeout(() => {
                state.notifications = state.notifications.filter((notif) => notif.id != newNotification.id)
            }, 5000)
        },
        [Mutations.ADD_USER](state, user: IUser) {
            state.users.push(user)
        },
        [Mutations.LOGIN_USER](state, token) {
            state.userLogged.token = token
            state.isLogged = true
        },
        [Mutations.USER_LOGGED](state, data) {
            state.userLogged.name = data.name
            state.userLogged.apiKey = data.apiKey
        },
    },
    actions: {
        /**
         * @name REGISTER_USER
         * @descripton efetua o registro do usuário e o adiciona a store
         * @param {IUser} user usuário com name, email e password
         */
        async [Actions.REGISTER_USER]({ commit }, user: IUser) {
            try {
                const response = await callApiClient.user.register(user.name, user.email, user.password)
                commit(Mutations.ADD_USER, response)
            } catch (e) {
                throw new Error(`Não foi possível efetuar cadastro. Confira os dados informados`)
            }
        },

        /**
         * @name LOGIN_USER
         * @description Efetua login do usuário, salva o token na store,
         * redireciona para a página de 'feedbacks' e aciona a action
         * para pegar os dados do usuário logado.
         * @param {IUser} user usuário com name, email e password
         */
        async [Actions.LOGIN_USER]({ commit, dispatch, state }, user: IUser) {
            try {
                const response = await callApiClient.user.login(user.email, user.password)
                commit(Mutations.LOGIN_USER, response.token)
                router.push('/feedbacks')
                dispatch(Actions.GET_USER, state.userLogged.token)
            } catch (e) {
                const { notify } = useNotifier()
                notify(TypeOfNotification.FALHA, 'Não foi possível efetuar login', 'Confira os dados informados.')
                throw new Error(`Não foi possível efetuar login. Confira os dados informados`)
            }
        },

        /**
         * @name GET_USER
         * @descripton Captura os dados do usuários e salva na store
         * @param token chave de acesso do usuário
         */
        async [Actions.GET_USER]({ commit }, token) {
            try {
                const response = await callApiClient.user.show(token)
                commit(Mutations.USER_LOGGED, response)
            } catch (e) {
                throw new Error(`Não foi possível capturar os dados do usuário.`)
            }
        },
    },
    modules: {},
})
