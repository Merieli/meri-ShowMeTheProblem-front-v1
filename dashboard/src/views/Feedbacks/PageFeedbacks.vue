<template>
    <div class="feedbacks">
        <GeralHeader />
        <section class="feedbacks__info text-brand-black">
            <h2 class="text-5xl font-black mb-2">Feedbacks</h2>
            <p class="font-normal text-base">Detalhes de todos os feedbacks recebidos.</p>
        </section>
        <div class="feedbacks__all mt-8 mx-8">
            <aside class="feedbacks__listing mr-16 min-w-[20%]">
                <h4 class="text-4xl font-black mb-9">Listagem</h4>
                <suspense>
                    <FeedbackFilters
                        @select="fetchFeedbacks"
                        class="animate__animated animate__fadeIn animate__faster"
                    />
                    <template #fallback>
                        <FiltersLoading class="mt-8" />
                    </template>
                </suspense>
            </aside>
            <section class="feedbacks__cards mt-12">
                <div v-if="hasErrors" class="text-lg text-center text-gray-800 font-regular">
                    Aconteceu um erro ao carregar os Feedbacks ðŸ˜¥
                </div>
                <div
                    v-if="!isLoading && !state.isLoadingFeedbacks && !feedbacks.length && !hasErrors"
                    class="text-lg text-center text-gray-800 font-regular"
                >
                    Ainda nenhum feedback recebido ðŸ¤“
                </div>
                <FeedbackCardLoading
                    v-if="(isLoading || state.isLoadingFeedbacks) && !hasErrors && !state.isLoadingMoreFeedbacks"
                />
                <FeedbackCard
                    v-else-if="!hasErrors"
                    v-for="(feedback, index) in feedbacks"
                    :key="feedback.id"
                    :is-opened="index === 0"
                    :feedback="feedback"
                />
                <FeedbackCardLoading
                    v-if="state.isLoadingMoreFeedbacks && feedbacks.length < paginationTotal"
                    :less="true"
                />
            </section>
        </div>
    </div>
</template>

<script lang="ts">
import FeedbackCard from '../../components/Feedbacks/FeedbackCard.vue'
import FeedbackCardLoading from '../../components/Feedbacks/FeedbackCardLoading.vue'
import FeedbackFilters from '../../components/Feedbacks/FeedbackFilters.vue'
import FiltersLoading from '../../components/Feedbacks/FiltersLoading.vue'
import GeralHeader from '../../components/GeralHeader.vue'
import { TFeedback } from '../../interfaces'
import { useStore } from '../../store'
import { Actions } from '../../store/type-actions'
import { computed, defineComponent, onErrorCaptured, onMounted, onUnmounted, reactive, ref } from 'vue'

export default defineComponent({
    name: 'PageFeedbacks',
    components: {
        GeralHeader,
        FeedbackFilters,
        FiltersLoading,
        FeedbackCard,
        FeedbackCardLoading,
    },
    setup() {
        const store = useStore()
        const hasErrors = computed(() => store.getters.hasErrors)
        const feedbacks = computed(() => store.getters.feedbacks)
        const currentFeedbackType = computed(() => store.getters.feedbackCurrentType)

        const paginationLimit = computed(() => store.getters.paginationFeedbacksLimit)
        const paginationOffset = computed(() => store.getters.paginationFeedbacksOffset)
        const paginationTotal = computed(() => store.getters.paginationFeedbacksTotal)

        const state = reactive({
            isLoadingFeedbacks: false,
            isLoadingMoreFeedbacks: false,
        })

        const hasErrorComponent = ref(false)

        function handleErrors(error: unknown) {
            state.isLoadingFeedbacks = false
            state.isLoadingMoreFeedbacks = false
            hasErrorComponent.value = !!error
        }

        const setTheCorrectParameters = (type?: TFeedback | string, offset?: number) => {
            const isNewTypeFeedback = type != currentFeedbackType.value
            let typeCorrect: Omit<TFeedback, 'all'> | ''
            let offsetCorrect: number

            typeCorrect = type ? type : currentFeedbackType.value
            offsetCorrect = offset ? offset : paginationOffset.value

            if (typeCorrect === 'all') {
                typeCorrect = ''
            }

            if (isNewTypeFeedback) {
                offsetCorrect = 0
                state.isLoadingMoreFeedbacks = false
            }

            return {
                typeCorrect,
                offsetCorrect,
            }
        }

        async function fetchFeedbacks(type?: TFeedback | string, offset?: number) {
            try {
                state.isLoadingFeedbacks = true
                const { typeCorrect, offsetCorrect } = setTheCorrectParameters(type, offset)
                return await store.dispatch(Actions.GET_ALL_FEEDBACKS, {
                    type: typeCorrect,
                    limit: paginationLimit.value,
                    offset: offsetCorrect,
                })
            } catch (e) {
                handleErrors(e)
            } finally {
                state.isLoadingFeedbacks = false
            }
        }

        async function handleScroll() {
            const isBottomOfWindow =
                Math.ceil(document.documentElement.scrollTop + window.innerHeight) >=
                document.documentElement.scrollHeight

            if (state.isLoadingFeedbacks || state.isLoadingMoreFeedbacks) return
            if (feedbacks.value.length >= paginationTotal.value) {
                state.isLoadingMoreFeedbacks = false
                return
            }
            if (!isBottomOfWindow) return

            const newPage = paginationOffset.value + paginationLimit.value
            try {
                state.isLoadingMoreFeedbacks = true
                fetchFeedbacks(currentFeedbackType.value, newPage)
            } catch (e) {
                handleErrors(e)
            }
        }

        onMounted(() => {
            if (feedbacks.value.length == 0) {
                fetchFeedbacks('all', 0)
            }
            window.addEventListener('scroll', handleScroll, false)
        })
        onUnmounted(() => {
            window.removeEventListener('scroll', handleScroll, false)
        })

        onErrorCaptured(handleErrors)

        return {
            feedbacks,
            state,
            isLoading: computed(() => store.getters.isLoading),
            hasErrors,
            fetchFeedbacks,
            paginationTotal,
        }
    },
})
</script>

<style lang="scss" scoped>
.feedbacks {
    &__info {
        background-color: theme('colors.brand.gray');
        text-align: center;
        padding: 6.43rem;
    }
    &__all {
        display: flex;
    }

    &__cards {
        width: 100%;
    }

    &__listing {
        width: 300px;
        height: 100%;
    }

    &__item {
        background-color: theme('colors.brand.gray');
        border-radius: 0.25rem;
        padding: 1.1rem 1.5rem;
        margin-bottom: 1.5rem;
        width: 100%;
    }
}
</style>
