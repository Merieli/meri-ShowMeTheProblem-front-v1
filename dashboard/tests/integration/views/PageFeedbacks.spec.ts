import { StateStoreShape } from '@/interfaces'
import { key } from '@/store'
import { Actions } from '@/store/type-actions'
import PageFeedbacks from '@/views/Feedbacks/PageFeedbacks.vue'
import { mount, VueWrapper } from '@vue/test-utils'
import { nextTick } from 'vue'
import { createStore, Store } from 'vuex'

describe('PageFeedback.vue', () => {
    let wrapper: VueWrapper

    const getters = {
        hasErrors: () => mockStore.state.hasErrors,
        feedbacks: () => mockStore.state.feedbacks,
        feedbackFilters: () => mockStore.state.feedbacks.filters,
        feedbackCurrentType: () => mockStore.state.feedbacks.currentType,
        paginationFeedbacksLimit: () => mockStore.state.feedbacks.pagination.limit,
        paginationFeedbacksOffset: () => mockStore.state.feedbacks.pagination.total,
        paginationFeedbacksTotal: () => mockStore.state.feedbacks.pagination.offset,
    }

    const actions = {
        [Actions.GET_ALL_FEEDBACKS]: jest.fn(),
        [Actions.GET_SUMMARY_FEEDBACK]: jest.fn(),
        [Actions.CHANGE_ACTIVE_FEEDBACK]: jest.fn(),
    }

    type TStoreState = Pick<StateStoreShape, 'feedbacks' | 'filters' | 'hasErrors' | 'isLoading'>

    const mockStore: Store<TStoreState> = createStore<TStoreState>({
        state: {
            feedbacks: {
                data: [], //o resultado depende da Action portanto nÃ£o deve ser mockado
                filters: [
                    {
                        label: 'Todos',
                        color: {
                            text: 'text-info',
                            bg: 'bg-info',
                        },
                        amount: 3,
                        active: true,
                        type: 'all',
                    },
                    {
                        label: 'Problemas',
                        color: {
                            text: 'text-danger',
                            bg: 'bg-danger',
                        },
                        amount: 2,
                        active: false,
                        type: 'issue',
                    },
                    {
                        label: 'Ideias',
                        color: {
                            text: 'text-warning',
                            bg: 'bg-warning',
                        },
                        amount: 1,
                        active: false,
                        type: 'idea',
                    },
                    {
                        label: 'Outros',
                        color: {
                            text: 'text-graydark',
                            bg: 'bg-graydark',
                        },
                        amount: 0,
                        active: false,
                        type: 'other',
                    },
                ],
                pagination: {
                    limit: 4,
                    offset: 0,
                    total: 0,
                },
                currentType: 'all',
            },
            filters: {
                all: 3,
                idea: 1,
                issue: 2,
                other: 0,
            },
            isLoading: false,
            hasErrors: false,
        },
        getters,
        actions,
    })

    beforeEach(() => {
        wrapper = mount(PageFeedbacks, {
            global: {
                plugins: [[mockStore, key]],
                stubs: ['GeralHeader'],
            },
        })
    })

    afterEach(() => {
        wrapper.unmount()
        jest.clearAllMocks()
    })

    describe('IntegraÃ§Ã£o/Componente', () => {
        describe('ðŸ‘€ RenderizaÃ§Ã£o:', () => {
            test('Dado o componente Quando renderizado EntÃ£o deve possuir html equivalente ao Snapshot gravado', () => {
                expect(wrapper.html()).toMatchSnapshot()
            })

            test('Dado o carregamento da pÃ¡gina Quando os filtros forem exibidos EntÃ£o deve haver os filtros "Todos, Problemas, Ideias e Outros"', () => {
                const filterLabels = wrapper.findAll('[data-filter="label"]')

                expect(filterLabels[0].text()).toBe('Todos')
                expect(filterLabels[1].text()).toBe('Problemas')
                expect(filterLabels[2].text()).toBe('Ideias')
                expect(filterLabels[3].text()).toBe('Outros')
            })

            test('Dado o carregamento da pÃ¡gina Quando os filtros forem exibidos EntÃ£o devem possuir a quantidade correta de feedbacks para cada categoria', () => {
                const filterAmount = wrapper.findAll('[data-filter="amount"]')

                expect(filterAmount[0].text()).toBe('3')
                expect(filterAmount[1].text()).toBe('2')
                expect(filterAmount[2].text()).toBe('1')
                expect(filterAmount[3].text()).toBe('0')
            })

            test('Dado os filtros Quando forem exibidos EntÃ£o o filtro "Todos" deve estar selecionado', () => {
                const filterTodos = wrapper.find('[data-filter="Todos"]')
                expect(filterTodos.attributes('class')).toContain('feedbacks-filter__item--selected')
            })
        })
        describe('ðŸ§  Comportamento:', () => {
            // No componente eu testo:
            // Confirmar se o state Ã© renderizado
            //a mutaÃ§Ã£o Ã© confirmada?
            // ele tem a carga correta?
            // NÃƒO testar se o getter funciona, mas se o cÃ³digo do componente exibe o conteudo corretamente que Ã© chamado de um getter
            // Testar se a action Ã© disparada quando o evento ocorre, e se ela tem a carga correta

            test('Dado a pÃ¡gina de feedbacks Quando carregar EntÃ£o deve chamar a Action uma vez para exibir todos os feedbacks', () => {
                const getFeedbacks = actions[Actions.GET_ALL_FEEDBACKS]

                expect(getFeedbacks).toHaveBeenCalled()
                expect(getFeedbacks).toHaveBeenCalledTimes(1)
            })

            test('Dado um usuÃ¡rio Quando carregar a pÃ¡gina EntÃ£o deve chamar a Action uma vez para exibir todos os filtros', () => {
                const getFilters = actions[Actions.GET_SUMMARY_FEEDBACK]

                expect(getFilters).toHaveBeenCalled()
                expect(getFilters).toHaveBeenCalledTimes(1)
            })

            test('Dado os filtros de feedbacks Quando clicar para exibir "Problemas" EntÃ£o deve mudar os feedbacks exibidos para apenas os dessa categoria chamando a Action uma vez', async () => {
                const activeFeedback = actions[Actions.CHANGE_ACTIVE_FEEDBACK]
                const filterIssue = wrapper.find('[data-filter="Problemas"]')
                await filterIssue.trigger('click')

                expect(activeFeedback).toHaveBeenCalled()
                expect(activeFeedback).toHaveBeenCalledTimes(1)
            })

            test('Dado o carregamento dos feedbacks Quando ocorrer algum erro EntÃ£o deve exibir a mensagem de erro', async () => {
                mockStore.state.hasErrors = true
                await nextTick()
                const errorMessage = wrapper.find('[data-feedbacks="error"]')

                expect(errorMessage.text()).toContain('Aconteceu um erro ao carregar os Feedbacks')
            })
        })
    })
})
