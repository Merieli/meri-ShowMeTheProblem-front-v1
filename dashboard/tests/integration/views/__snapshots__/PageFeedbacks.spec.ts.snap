// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFeedback.vue IntegraÃ§Ã£o/Componente ðŸ‘€ RenderizaÃ§Ã£o: Dado o componente Quando renderizado EntÃ£o deve possuir html equivalente ao Snapshot gravado 1`] = `
"<div class=\\"feedbacks\\">
  <geral-header-stub></geral-header-stub>
  <section class=\\"feedbacks__info text-brand-black\\">
    <h2 class=\\"text-5xl font-black mb-2\\">Feedbacks</h2>
    <p class=\\"font-normal text-base\\">Detalhes de todos os feedbacks recebidos.</p>
  </section>
  <div class=\\"feedbacks__all mt-8 mx-8\\">
    <aside class=\\"feedbacks__listing mr-16 min-w-[20%]\\">
      <h4 class=\\"text-4xl font-black mb-9\\">Listagem</h4>
      <div class=\\"feedbacks-filter animate__animated animate__fadeIn animate__faster\\">
        <h5 class=\\"font-bold mb-5 text-2xl\\">Filtros</h5>
        <ul class=\\"feedbacks-filter__items\\">
          <li class=\\"feedbacks-filter__item animate__animated animate__fadeIn animate__faster feedbacks-filter__item--selected\\" data-filter=\\"Todos\\"><span data-filter=\\"label\\"><i class=\\"bg-info feedbacks-filter__circle\\"></i> Todos</span><span class=\\"text-info font-bold\\" data-filter=\\"amount\\">3</span></li>
          <li class=\\"feedbacks-filter__item animate__animated animate__fadeIn animate__faster\\" data-filter=\\"Problemas\\"><span data-filter=\\"label\\"><i class=\\"bg-danger feedbacks-filter__circle\\"></i> Problemas</span><span class=\\"text-brand-graydark font-bold\\" data-filter=\\"amount\\">2</span></li>
          <li class=\\"feedbacks-filter__item animate__animated animate__fadeIn animate__faster\\" data-filter=\\"Ideias\\"><span data-filter=\\"label\\"><i class=\\"bg-warning feedbacks-filter__circle\\"></i> Ideias</span><span class=\\"text-brand-graydark font-bold\\" data-filter=\\"amount\\">1</span></li>
          <li class=\\"feedbacks-filter__item animate__animated animate__fadeIn animate__faster\\" data-filter=\\"Outros\\"><span data-filter=\\"label\\"><i class=\\"bg-graydark feedbacks-filter__circle\\"></i> Outros</span><span class=\\"text-brand-graydark font-bold\\" data-filter=\\"amount\\">0</span></li>
        </ul>
      </div>
    </aside>
    <section class=\\"feedbacks__cards mt-12\\">
      <div class=\\"text-lg text-center text-gray-800 font-regular\\" data-feedbacks=\\"empty\\"> Ainda nenhum feedback recebido ðŸ¤“ </div>
      <!--v-if-->
    </section>
  </div>
</div>"
`;
